### PR TITLE
Improve `MRAE` for DDP

### PR DESCRIPTION
**Pull request by KickItLikeShika**
Improving `MedianRelativeAbsoluteError` implementation to be compatible with DDP and add the necessary tests.

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by KickItLikeShika** [comment]
There is CI error related to `WaveHedgesDistance` https://github.com/pytorch/ignite/pull/1891/checks?check_run_id=2208768289#step:9:259 , I will send another PR to fix that

File: tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py [hunk 1]
```diff
@@ -102,3 +105,174 @@ def update_fn(engine, batch):
...
+        e = np.abs(np_y_true - np_y_preds) / np.abs(np_y_true - np_y_true.mean())
+        np_res = np.median(e)
+
+        e_prepend = np.insert(e, 0, e[0], axis=0)
+        np_res_prepend = np.median(e_prepend)
+
+        # The results between numpy.median() and torch.median() are Inconsistant
+        # when the length of the array/tensor is even. So this is a hack to avoid that.
+        # issue: https://github.com/pytorch/pytorch/issues/1837
+        if np_y_preds.shape[0] % 2 == 0:
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py:196]: Total size is 151 here, I think we do not need this trick here ?

**Review by vfdev-5 — COMMENTED** [review]
Thanks @KickItLikeShika , I left a comment

File: tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py (hunk 1 above)
Comment (KickItLikeShika) [reply tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py:196]: Yes we are not gathering here, thanks for this!

**Comment by KickItLikeShika** [comment]
@vfdev-5 there are some tests need to be restarted because of connection errors

**Review by sdesrozis — APPROVED** [review]
@KickItLikeShika Thank you! LGTM

File: tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py [hunk 2]
```diff
@@ -102,3 +105,165 @@ def update_fn(engine, batch):
...
+        y_pred = idist.all_gather(y_pred)
+        y = idist.all_gather(y)
+
+        np_y_pred = y_pred.cpu().numpy().ravel()
+        np_y = y.cpu().numpy().ravel()
+
+        res = m.compute()
+
+        e = np.abs(np_y - np_y_pred) / np.abs(np_y - np_y.mean())
+        np_res = np.median(e)
```
Comment (sdesrozis) [inline tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py:135]: This line should be moved in next `else` statement (L145)

File: tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py [hunk 3]
```diff
@@ -102,3 +105,165 @@ def update_fn(engine, batch):
...
+
+        e_prepend = np.insert(e, 0, e[0], axis=0)
```
Comment (sdesrozis) [inline tests/ignite/contrib/metrics/regression/test_median_relative_absolute_error.py:137]: These 2 lines should be moved in the next `if` statement (L144)

**Comment by sdesrozis** [comment]
@KickItLikeShika left 2 minor comments

**Pull request merged**